docs/verifier/query-builder.md

```diff
@@ -2,59 +2,209 @@
+- **Brute-force Resistance:** Prevents brute-forcing to reveal Genesis ID or other values.
+- **Tracking Resistance:** Ensures Profile IDs and Genesis ID cannot be linked based on the Sybil ID.
+- **Flexible Session Duration:** Allows the verifier to keep the session active as needed, without time-specific constraints.
+- **Protection Against Collusion:** Prevents matching different identifiers as belonging to the same identity, even if the issuer and verifier collude.
+
+### Query Type
+
+The protocol allows selection between two query types:
+
+1. **Condition:** This type allows the prover to demonstrate that a specific condition is met without revealing the underlying data. It maintains data privacy by cryptographically verifying the condition. An example is proving legal age for entry into a venue without disclosing the actual age.
+
+2. **Selective disclosure:** Use this when the prover needs to reveal specific data stored in their credential. This option discloses the actual value of an attribute to the verifier. For instance, instead of just proving legal age, it would show the verifier the prover's actual age.
+
+Choose "Condition" for privacy-preserving verification, or "Selective disclosure" for cases where revealing specific information is necessary.
+
+### Operator
+
+In conditional queries, the "Operator" input defines the logical operation to be applied. The supported operators are:
+
+- **Equal:** Checks if the value is equal to the specified value.
+- **Not Equal:** Verifies if the value is not equal to the specified value.
+- **Less Than:** Assesses if the value is less than the given value.
+- **Greater Than:** Evaluates if the value is greater than the specified value.
+- **Matches one of the values:** Determines if the value is within a given set of values.
+- **Matches none of the values:** Checks if the value is not within a specified set.
 
-Then, one can copy and paste the snippet on the verifier API `GetAuthRequest()` function.
+Choose the appropriate operator based on the condition you need to validate.
 
-:::info
+### Attribute value
 
-The idea behind the Query Builder is to facilitate the query elaboration process by generating the whole section relative to "Request for specific proof" of the auth request endpoint code, as it can be seen on the [<ins>Run a Verifier tutorial</ins>](/docs/verifier/verification-library/verifier-set-up.md#verifier-server-setup), more specifically on point 3.
+The "Attribute value" field allows users to input the specific value or range of values to be compared with the selected attribute in the prover's credential. This field adapts to the type of the chosen attribute, accommodating various data types such as strings, numbers, booleans, dates, and more. This ensures that the value provided is compatible with the selected operator and the nature of the attribute.
```

Comment:
> The "Attribute value" field allows users to input the specific value or range of values *used to* compare with the selected attribute in the *user's* credential.

The prover just takes the required data from the credential and the query as an input and it outputs the proof. The credentials always belong to the users.

Reply:
> The prover just takes the required data from the credential and the query as an input and it outputs the proof. The credentials always belong to the users.

Man, what a mess! In your sentence above the `prover` and the `user` are the same, right? I mean the prover is not software, it's the person that (using software) proves something.

Also, I think we have to avoid using the word "user". It's ambiguous because we use it both for the verifier and the prover (and maybe even the issuer...).

I'd rephrase to:

> The "Attribute value" field allows verifiers to input the specific value or range of values used to compare with the selected attribute in the prover's credential.

Reply:
Done